cql3/query_processor.cc

```diff
@@ -1002,6 +1002,24 @@ query_processor::forward(query::forward_request req, tracing::trace_state_ptr tr
     co_return co_await remote_.get().forwarder.dispatch(std::move(req), std::move(tr_state));
 }
 
+::shared_ptr<messages::result_message> result_with_workaround_for_concurrent_statements(
+        const statements::schema_altering_statement& stmt) {
+    using event_t = cql_transport::event::schema_change;
+    ::shared_ptr<event_t> ce;
+    // in theory we could add the workaround to more (or even all) schema statements
+    // but for now let's limit to what we know causes problems for naive clients
+    if (auto create_ks = dynamic_cast<const statements::create_keyspace_statement*>(&stmt)) {
+        ce = ::make_shared<event_t>(
+                event_t::change_type::CREATED,
+                event_t::target_type::KEYSPACE,
+                create_ks->keyspace());
+    }
```

Comment:
Should get the event from the statement itself rather than a dynamic case here, using a virtual function.

I see no reason not to support CREATE TABLE too.

Reply:
I can add it, but question is where to draw the line? see https://github.com/scylladb/scylladb/pull/18819#discussion_r1611044799


> dynamic cast here, using a virtual function

then a hack suddenly becomes part of statement interface, is this what we want?

Reply:
added